backend/auth/middleware.go

```diff
@@ -0,0 +1,148 @@
+			c.Next()
+			return
+		}
+		if c.Request.Method == "POST" && strings.HasSuffix(c.Request.URL.Path, "account") {
+			firstAccountIssued, err := IsFirstAccountIssued()
+			if err != nil {
+				logger.AuthLog.Errorln(err.Error())
+				c.JSON(http.StatusInternalServerError, gin.H{"error": "failed to authorize user account creation"})
+				c.Abort()
+				return
+			}
+			if !firstAccountIssued {
+				c.Next()
+				return
+			}
+		}
+		claims, err := getClaimsFromAuthorizationHeader(c.Request.Header.Get("Authorization"), jwtSecret)
+		if err != nil {
+			logger.AuthLog.Errorln(err.Error())
+			c.JSON(http.StatusUnauthorized, gin.H{"error": fmt.Sprintf("auth failed: %s", err.Error())})
+			c.Abort()
+			return
+		}
+		if claims.Permissions == USER_ACCOUNT && strings.HasPrefix(c.Request.URL.Path, "/config/v1/account") {
+			requestAllowed, err := requestAllowedForRegularUser(claims, c.Request.Method, c.Request.URL.Path)
+			if err != nil {
+				c.JSON(http.StatusInternalServerError, gin.H{"error": "failed to authorize operation"})
+				c.Abort()
+				return
+			}
+			if !requestAllowed {
+				c.JSON(http.StatusForbidden, gin.H{"error": "forbidden"})
+				c.Abort()
+				return
+			}
+		}
+		c.Next()
+	}
+}
+
+func getClaimsFromAuthorizationHeader(header string, JwtSecret []byte) (*jwtGocertClaims, error) {
+	if header == "" {
+		return nil, fmt.Errorf("authorization header not found")
+	}
+	bearerToken := strings.Split(header, " ")
+	if len(bearerToken) != 2 || bearerToken[0] != "Bearer" {
+		return nil, fmt.Errorf("authorization header couldn't be processed. The expected format is 'Bearer token'")
+	}
+	claims, err := getClaimsFromJWT(bearerToken[1], JwtSecret)
+	if err != nil {
+		return nil, fmt.Errorf("token is not valid")
+	}
+	return claims, nil
+}
+
+func getClaimsFromJWT(bearerToken string, JwtSecret []byte) (*jwtGocertClaims, error) {
+	claims := jwtGocertClaims{}
+	token, err := jwt.ParseWithClaims(bearerToken, &claims, func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
+		}
+		return JwtSecret, nil
+	})
+	if err != nil || !token.Valid {
+		return nil, err
+	}
+	return &claims, nil
+}
+
+func requestAllowedForRegularUser(claims *jwtGocertClaims, method, path string) (bool, error) {
+	allowedPaths := []struct {
+		method, pathRegex string
+	}{
+		{"GET", `/config/v1/account\/(\w+)$`},
+		{"POST", `/config/v1/account\/(\w+)\/change_password$`},
+	}
+	for _, pr := range allowedPaths {
+		regex, err := regexp.Compile(pr.pathRegex)
+		if err != nil {
+			return false, fmt.Errorf("regex couldn't compile: %s", err)
+		}
+		matches := regex.FindStringSubmatch(path)
+		if len(matches) > 0 && method == pr.method {
+			if matches[1] == claims.Username {
+				return true, nil
+			}
+			return false, nil
+		}
+	}
+	return false, nil
+}
```

Comment:
I would discourage from having complicated logic here as to what is allowed by who, instead let's have this in `routers.go` right next to each handler